external_search.py

```diff
@@ -704,74 +672,230 @@ def _collections():
             _collections
         )
 
-class ExternalSearchIndexVersions(object):
 
-    VERSIONS = ['v4']
+class MappingDocument(object):
```

Comment:
With this class taking over most of the job of building the mapping document, it's possible to write some finer-grained unit tests, but I haven't done that and I don't know if it's valuable -- the proof will always be whether the tests against a real server find the things we want it to find.